### PR TITLE
Changes after code review

### LINKED ISSUE #1: Code review
Hi Olivier,

As requested, some feedback on the code can be found below.

> if (info.menuItemId == 14) { // 14 is the index for LinkedIn

and
> searchEngines = sortAlphabetically(data);
>            var index = 0;
>            for (var se in searchEngines) {
>var strId = index.toString();

Please don't use menuItemId to check whether this is LinkedIn. This id can change at all times. Instead, use the key (or name property) from the JSON file as an id, that will work as well. 

Also note that due a bug with info.selectionText - https://bugzilla.mozilla.org/show_bug.cgi?id=1338898 - you probably should implement a workaround for that. Workaround in Translate Now:
> sendMessage("getSelection", selectionText, priviledgedSiteNoContentScript);

For all code on this matter, checkout https://github.com/Smile4ever/firefoxaddons/blob/master/Translate%20Now-webext/background.js and https://github.com/Smile4ever/firefoxaddons/blob/master/Translate%20Now-webext/translatenow.js

>  contexts: ["all"]

Please don't do this, and consider only those contexts that make sense. Currently you have only implemented "selection" (because when you right click the page without selection, nothing happens!)

>  var creating = browser.tabs.create({
>        url: targetUrl
>    });
>creating.then(null, onError);

I would shorten this to
>  browser.tabs.create({
>        url: targetUrl
>    }).then(null, onError);

or even

>  browser.tabs.create({ url: targetUrl });

Keep up the good work!

### PR DESCRIPTION
**Pull request by Smile4ever**
* Use info.selectionText for lengths <150 and lengths >150. This is more reliable. The content script should only be a fallback.
* Simplify a bit.
* Implement selected text for input and textareas.

Fixes #1

### CODE REVIEW

**Comment by odebroqueville** [comment]
Hey Geoffrey,

In background.js:

info.selectionText doesn’t work if length >=150 (everything beyond 149 characters gets truncated) so I don’t understand why you include the condition || info.selectionText.length > 150 on line 61

Olivier
P.S. I think I’m starting to get the hang of git and github! It’s really cool!

**Comment by Smile4ever** [comment]
> info.selectionText doesn’t work if length >=150 (everything beyond 149 characters gets truncated) so I don’t understand why you include the condition || info.selectionText.length > 150 on line 61

I can understand you being confused. On recent Firefox versions, this "150 char limit bug" is fixed. So if you get more than 150 characters, you are running a recent Firefox version without this bug (no workaround required).

If you get 149 characters or less, this workaround also isn't needed (since your selection is not longer than 150 characters = you don't hit the bug).

Only when your selection is exactly 150 characters long, I get suspicious. This user is probably running an old version of Firefox with this bug or just happens to be selecting exactly 150 characters. For the first case, you will want to use the workaround code (in the content script).

For the second case, it doesn't matter if we're running the workaround. The workaround is good enough, so it probably won't break 👍 .

**Comment by odebroqueville** [comment]
On 27 Aug 2017, 19:30 +0200, Geoffrey De Belie <notifications@github.com>, wrote:
> > info.selectionText doesn’t work if length >=150 (everything beyond 149 characters gets truncated) so I don’t understand why you include the condition || info.selectionText.length > 150 on line 61
> I can understand you being confused. On recent Firefox versions, this "150 char limit bug" is fixed. So if you get more than 150 characters, you are running a recent Firefox version without this bug (no workaround required).
I tested on ffx 55.0.3 (latest stable release) and it truncated everything beyond 150, so I think I should remove the condition > 150 to remain on the safe side.

**Comment by Smile4ever** [comment]
> I tested on ffx 55.0.3 (latest stable release) and it truncated everything beyond 150, so I think I should remove the condition > 150 to remain on the safe side.

info.selectionText will indeed return 150 characters, so it will use the "selection" value which has more than 150 characters. See the code :) 

> Good thinking and nice work, Geoffrey. Will definitely add your name as a contributor to this extension. Thanks

Thank you!
